src/components/DateInput/dateInput.test.js

```diff
@@ -54,88 +56,84 @@ describe('DateInput component', () => {
-      input = null;
     });
 
-    it('calls onInputChange when the input value changes', () => {
+    it('calls onInputChange when the input value changes', async () => {
       expect(inputChangeFn).not.toHaveBeenCalled();
-      input.simulate('change', { target: { value: '4' } });
+      await userEvent.type(screen.getByRole('textbox'), '4')
       expect(inputChangeFn).toHaveBeenCalled();
     });
 
-    it('calls onDateChange when user types a VALID freeform date', () => {
+    it('calls onDateChange when user types a VALID freeform date', async () => {
       expect(dateChangeFn).not.toHaveBeenCalled();
-      input.simulate('change', { target: { value: '4/20/2020' } });
+      await userEvent.type(screen.getByRole('textbox'), '4/20/2020')
       expect(dateChangeFn).toHaveBeenCalled();
     });
 
-    it('does not call onDateChange when user types an INCOMPLETE freeform date', () => {
+    it('does not call onDateChange when user types an INCOMPLETE freeform date', async () => {
       expect(dateChangeFn).not.toHaveBeenCalled();
-      input.simulate('change', { target: { value: '2/3' } });
+      await userEvent.type(screen.getByRole('textbox'), '2/3')
       expect(dateChangeFn).not.toHaveBeenCalled();
     });
 
-    it('does not call onDateChange when user types an INVALID freeform date', () => {
+    it('does not call onDateChange when user types an INVALID freeform date', async () => {
       expect(dateChangeFn).not.toHaveBeenCalled();
-      input.simulate('change', { target: { value: '2/30/2020' } });
+      await userEvent.type(screen.getByRole('textbox'), '2/30/2020')
       expect(dateChangeFn).not.toHaveBeenCalled();
     });
 
     it('calls onDateChange when user selects a day in the picker', () => {
       expect(dateChangeFn).not.toHaveBeenCalled();
-      wrapper.find('Day').at(7).simulate('click');
+      fireEvent.click(screen.getByText('7'));
       expect(dateChangeFn).toHaveBeenCalled();
     });
 
     it('updates inupt value when user selects a day in the picker', () => {
-      wrapper.find('Day').at(7).simulate('click');
-      expect(input.instance().value).toHaveLength(10); // MM/DD/YYYY === 10 chars
+      fireEvent.click(screen.getByText('7'));
+      expect(screen.getByRole('textbox').value).toBe('07/07/2020'); // MM/DD/YYYY === 10 chars
+      expect(screen.getByRole('textbox').value).toHaveLength(10); // MM/DD/YYYY === 10 chars
     });
 
-    it('sets DayPicker selected date correctly from input', () => {
-      input.simulate('change', { target: { value: '4/20/2020' } });
-      const dayPickerDate = wrapper.find('DayPicker').prop('month');
-      expect(dayPickerDate.getMonth()).toBe(3);
-      expect(dayPickerDate.getFullYear()).toBe(2020);
+    it('sets DayPicker selected date correctly from input', async () => {
+      await userEvent.type(screen.getByRole('textbox'), '4/20/2020')
+      expect(screen.getByRole('gridcell', { selected: true, name: 'Mon Apr 20 2020' })).toBeTruthy()
```

Comment:
The previous way just finds a prop. This finds the "actual" selection in question by it's role, label text, and then finally that it's selected (all the other options are unselected except this one)